First draft of how `experimental` should be used. (#1999)
oscar-system/Oscar.jl: docs/src/DeveloperDocumentation/experimental.md

# Implementing features from scratch

> For new features whose interface naturally will change a lot during the

Reviewer: ```suggestion
For new features whose interface is likely to change a lot during the
```

implementation phase, Oscar.jl has a staging area, the `experimental` folder.

1. The new feature is implemented in the `experimental` folder.

> 2. Additionally to the original authors, a maintainer is assigned to guide the

Reviewer: This would be something new, so far we haven't done it. I see some merit (we want to ensure some forward progress happens and that someone takes responsibility). But for e.g. people who are already active (e.g. @fieker has quite some in `experimental`, I am not sure assigning a second internal person would be helpful?

Reviewer 2: I just wanted to write: We need a second person only for newbies.

But maybe that's not true.  Maybe even OSCAR veterans could occasionally discuss why some of their code is sitting in experimental for a quite a while?

The main problem with the current version of `experimental` is that the user cannot really distinguish from the actual core.

Reviewer: Indeed. Some options come to mind:
- maybe accessing code in experimental should be "guarded" and require `using Oscar.Experimental`, or require writing `Experimantal.FOO` (so `Experimental.galois_group(...)`
  - pro: obviously experimental
  - con: less discoverable
- or maybe the primary functions should have a `@warn_experimental()` call in them, which prints a warning about them being experimental (possibly only the first time they are called)
- ... other ideas?

Author: For the discoverability: I still think, code should not spend too much time in `experimental`, then this would matter less. I don't know whether we want to provide a concrete time frame though.

Reviewer 3: Code in experimental will be buggy because it is less tested and not battle proof yet. So the naive user should not use it at all. Certainly not to prove a theorem. It is like a preprint which has not been peer reviewed yet - use at your own risk.
